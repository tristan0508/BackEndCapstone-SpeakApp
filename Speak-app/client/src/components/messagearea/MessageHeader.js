import React, { useState } from 'react';
import {
    Divider,
    makeStyles,
    createStyles,
    AppBar,
    Toolbar,
    IconButton,
    ThemeProvider,
    Tooltip,
    ButtonBase,
    Avatar
} from '@material-ui/core';
import AssistantOutlinedIcon from '@material-ui/icons/AssistantOutlined';
import { headerTheme } from '../../customtheme/MaterialTheme';
import { AvatarGroup } from '@material-ui/lab';
import avatar from '../../images/headshot.jpg';
import MenuOpenIcon from '@material-ui/icons/MenuOpen';


const useStyles = makeStyles(() =>
    createStyles({
        grow: {
            flexGrow: 1,
        },
        topic: {

        },
        active: {
            backgroundColor: '#36454f',
            boxShadow: 'inset 0px 0px 5px #c1c1c1',
            outline: 'none'
        }
    }),
);


const MessageHeader = ({ openMenu, setOpenMenu }) => {
    const classes = useStyles();

    const [open, setOpen] = useState(false)
    const [initWidth, setInitWidth] = useState(0)


    const handleOpen = () => {
        if (!open) {
            setOpen(true);
        } else {
            setOpen(false);
        }
    }


    const handleOpenMenu = () => {
        
        if (!openMenu) {
            setOpenMenu(true)
        } else {
            setOpenMenu(false)
        }
      
        handleResize();
    }

    window.onresize = function(){
        let box = document.getElementById('chat-grid-container');
        let width1 = box.offsetWidth;
        setInitWidth(width1)
        if(width1 !== initWidth && openMenu === true){
            document.documentElement.style.setProperty('--msg-width', `${width1}px`)

        }
        
    }
    
    const handleResize = () => {
        let box = document.getElementById('chat-grid-container');
        let box1 = document.getElementById('side-container');
        let width1 = box.offsetWidth;
        let width2 = box1.offsetWidth;


        openMenu? document.documentElement.style.setProperty('--msg-width', '100%') : 
        document.documentElement.style.setProperty('--msg-width', `${width1 - width2}px`)
    }

    return (
        <ThemeProvider theme={headerTheme}>
            <AppBar position="relative" style={{ backgroundColor: '#1dcaff', border: '1px solid white' }}>
                <Toolbar>
                    <IconButton color="inherit" title="Pin">
                        <AssistantOutlinedIcon />
                    </IconButton>
                    <Divider orientation="vertical" flexItem variant="middle" />
                    <ButtonBase>
                        <Tooltip
                            onClick={handleOpen}
                            open={open}
                            title="Add"
                            color="inherit"
                        >
                            <div>Topic</div>
                        </Tooltip>

                    </ButtonBase>

                    <AvatarGroup className="avatar" max={2} title={`View all 3 members`}>
                        <Avatar alt="Remy Sharp" src={avatar} />
                        <Avatar alt="Remy Sharp" src="/static/images/avatar/1.jpg" />
                        <Avatar alt="Remy Sharp" src="/static/images/avatar/1.jpg" />
                    </AvatarGroup>

                    <IconButton onClick={handleOpenMenu} color="inherit">
                        <MenuOpenIcon className={openMenu ? classes.active : undefined} />
                    </IconButton>
                </Toolbar>
            </AppBar>
        </ThemeProvider>
    )
}

export default MessageHeader;