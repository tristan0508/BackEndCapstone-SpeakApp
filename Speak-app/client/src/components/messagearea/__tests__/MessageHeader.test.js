import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import MessageHeader from '../MessageHeader';



describe('Message Header', () => {
    test('renders Message Header component', () => {
        const tree = renderer
        .create(<MessageHeader/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('handleOpen true or false', () => {
        const {getByTestId} = render(<MessageHeader />)
        expect(fireEvent.click(getByTestId('handleOpen'))).toBeCalled
    })

});

