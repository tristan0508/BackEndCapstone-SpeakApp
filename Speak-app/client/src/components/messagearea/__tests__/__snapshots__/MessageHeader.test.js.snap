// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Header renders Message Header component 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary MuiPaper-elevation4"
  style={
    Object {
      "backgroundColor": "#1dcaff",
      "border": "1px solid white",
    }
  }
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Pin"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h4l3 3 3-3h4c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 16h-4.83l-.59.59L12 20.17l-1.59-1.59-.58-.58H5V4h14v14zm-7-1l1.88-4.12L18 11l-4.12-1.88L12 5l-1.88 4.12L6 11l4.12 1.88z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <hr
      className="MuiDivider-root MuiDivider-middle MuiDivider-flexItem MuiDivider-vertical"
    />
    <button
      className="MuiButtonBase-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        aria-describedby={null}
        className=""
        color="inherit"
        data-testid="handleOpen"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title="Add"
      >
        Topic
      </div>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <div
      className="MuiAvatarGroup-root avatar"
      title="View all 3 members"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle MuiAvatarGroup-avatar"
        style={
          Object {
            "marginLeft": undefined,
            "zIndex": 3,
          }
        }
      >
        <img
          alt="Remy Sharp"
          className="MuiAvatar-img"
          src="headshot.jpg"
        />
      </div>
      <div
        className="MuiAvatar-root MuiAvatar-circle MuiAvatarGroup-avatar MuiAvatar-colorDefault"
        style={
          Object {
            "marginLeft": null,
            "zIndex": 0,
          }
        }
      >
        +
        2
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
      data-testid="handleMenu"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h13v-2H3v2zm0-5h10v-2H3v2zm0-7v2h13V6H3zm18 9.59L17.42 12 21 8.41 19.59 7l-5 5 5 5L21 15.59z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</header>
`;
